### PR TITLE
Integrate newer cardano-ledger and ouroboros-consensus

### PR DESCRIPTION
**Pull request by dnadales**
# Changelog

```yaml
- description: |
    - Add `getReferenceInputsSizeForTxIds` function.
    - Bump `cardano-ledger` and `ouroboros-consensus` bounds. In particular, the Conway genesis parameter `ucppMinFeeRefScriptCostPerByte` needs to be set to the correct value after benchmarking.
    - Add an extra parameter to `evaluateTransactionFee`, the reference script size, which the Ledger requires to estimate the minimum fee of a transaction.
# uncomment types applicable to the change:
  type:
    - feature
    - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Additional context for the PR goes here. If the PR fixes a particular issue please provide a [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=) to the issue.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs [hunk 1]
```diff
@@ -96,6 +96,6 @@ handleConwayNEWEPOCHEvents conwayNewEpochEvent =
                                                 (convertRetiredPoolsMap refundPools)
                                                 (convertRetiredPoolsMap unclaimedPools)
         Conway.SnapEvent _ -> Nothing
-    Conway.TotalAdaPotsEvent _ -> Nothing
-
+        Conway.GovInfoEvent _ _ _ -> undefined -- FIXME: Which ledger event do we need here?
```
Comment (carbolymer) [inline cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs:99]: I think we might need a new event here. I'm not sure about this one. @Jimbo4350 do you have any idea?

File: cardano-api/cardano-api.cabal [hunk 2]
```diff
@@ -1,7 +1,7 @@
 cabal-version: 3.4
 
 name:                   cardano-api
-version:                8.43.0.0
```
Comment (carbolymer) [inline cardano-api/cardano-api.cabal:4]: This is not needed. Version bump should go in a separate PR.

File: cardano-api/cardano-api.cabal [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+version:                8.44.0.0
```
Comment (dnadales) [inline cardano-api/cardano-api.cabal:4]: This should go in a separate PR

File: cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs [hunk 4]
```diff
@@ -96,6 +96,6 @@ handleConwayNEWEPOCHEvents conwayNewEpochEvent =
                                                 (convertRetiredPoolsMap refundPools)
                                                 (convertRetiredPoolsMap unclaimedPools)
         Conway.SnapEvent _ -> Nothing
-    Conway.TotalAdaPotsEvent _ -> Nothing
-
+        Conway.GovInfoEvent {} -> Nothing -- FIXME: Confirm if we need a new event here
```
Comment (dnadales) [inline cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs:99]: @carbolymer do we need to address this in this PR?

File: cardano-api/cardano-api.cabal [hunk 5]
```diff
@@ -194,22 +194,23 @@ library internal
                       , optparse-applicative-fork
                       , ouroboros-consensus ^>= 0.16
                       , ouroboros-consensus-cardano ^>= 0.14
-                      , ouroboros-consensus-diffusion ^>= 0.12
+                      -- TODO: ouroboros-consensus-diffusion needs to be bumped
```
Comment (Jimbo4350) [inline cardano-api/cardano-api.cabal]: Is this comment still relevant?

File: cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs]: I would return `Nothing`. I think we should stop using ledger events because they don't actually represent the ledger state.

File: cardano-api/internal/Cardano/Api/LedgerState.hs [hunk 6]
```diff
@@ -1165,6 +1165,8 @@ mkProtocolInfoCardano (GenesisCardano dnc byronGenesis shelleyGenesisHash transC
             }
       , Consensus.hardForkTriggers = ncHardForkTriggers dnc
       , Consensus.ledgerTransitionConfig = transCfg
+        -- FIXME: add a comment mentioning when the a RE should change 'checkpoints' (eg around Genesis release).
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/LedgerState.hs]: Is this comment still relevant?

File: cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs (hunk 4 above)
Comment (Jimbo4350) [reply cardano-api/internal/Cardano/Api/LedgerEvents/Rule/TICK/NEWEPOCH.hs:99]: This is fine. I actually want to stop using ledger events because they don't actually represent the ledger state.

File: cardano-api/internal/Cardano/Api/Tx/Body.hs [hunk 7]
```diff
@@ -3065,6 +3060,17 @@ genesisUTxOPseudoTxIn nw (GenesisUTxOKeyHash kh) =
              (Shelley.KeyHashObj kh)
              Shelley.StakeRefNull
 
+-- | Calculate the reference inputs size for provided set of transaction IDs and UTXOs.
+getReferenceInputsSizeForTxIds
+  :: forall era ledgerera. ShelleyLedgerEra era ~ ledgerera
```
Comment (Jimbo4350) [inline cardano-api/internal/Cardano/Api/Tx/Body.hs:3065]: Unnecessary existential quanification.

Comment (carbolymer) [reply cardano-api/internal/Cardano/Api/Tx/Body.hs]: Removed

**Review by Jimbo4350 — APPROVED** [review]
